src/Functions/dateDiff.cpp

```diff
@@ -167,9 +173,13 @@ class DateDiffImpl
     template <typename TransformX, typename TransformY, typename T1, typename T2>
     Int64 calculate(const TransformX & transform_x, const TransformY & transform_y, T1 x, T2 y, const DateLUTImpl & timezone_x, const DateLUTImpl & timezone_y) const
     {
+        UInt8 day_of_week_mode = enable_default_monday_first ? 0 : 3;
+
         if constexpr (is_diff)
-            return static_cast<Int64>(transform_y.execute(y, timezone_y))
-                - static_cast<Int64>(transform_x.execute(x, timezone_x));
+        {
+            return static_cast<Int64>(transform_y.execute(y, timezone_y, day_of_week_mode))
```

Comment:
timezone_y is ideally the last argument (please see my comment in toStartOfInterval.cpp)

Reply:
I have been able to do this for `toStartOfInterval`, but here it is a bit more difficult since all `toRelative${unit}Num` functions are exposing `ToRelative${unit}NumImpl` with some levels of abstraction. I'll give it a try.